python/faery/display.py

```diff
@@ -0,0 +1,150 @@
+import os
+import sys
+import typing
+
+from . import events_stream, frame_stream, timestamp
+
+
+def generate_progress_bar(width: int, progress: typing.Optional[float]) -> str:
```

Comment:
Hehe is this to avoid depending on tqdm?

Reply:
Indeed! We could use tqdm but this is not all that much code.

Reply:
I understand your aversion towards dependencies, and if this works, I'm all for it. But tqdm is, I would argue, a relatively innocent dependency. I would prefer to avoid as much overhead/maintenance as possible. But let's leave it at this for now :)